dashboard/app/models/peer_review.rb

```diff
@@ -46,15 +85,26 @@ def localized_status_description
     I18n.t("peer_review.#{status}.description").html_safe if status
   end
 
-  def self.create_for_submission(user_level, level_source_id, from_instructor = false)
+  def self.create_for_submission(user_level, level_source_id)
+    created = []
+
     REVIEWS_PER_SUBMISSION.times do
-      create!(
+      created << create!(
         submitter_id: user_level.user.id,
-        from_instructor: from_instructor,
+        from_instructor: false,
         script: user_level.script,
         level: user_level.level,
         level_source_id: level_source_id
-      )
+      ).id
     end
+
+    # Remove old unassigned reviews for this submitter+script+level combination
+    where(
+      reviewer_id: nil,
+      submitter_id: user_level.user.id,
+      from_instructor: false,
+      script: user_level.script,
+      level: user_level.level,
+    ).where.not(id: created).destroy_all
   end
```

Comment:
Any reason you don't do this first?

Also, consider wrapping this in a transaction?

begin transaction
-remove old reviews
-make new reviews
end transaction